### PR TITLE
jv - convert ownerId to ObjectId

### PR DESCRIPTION
**Pull request by jvoucsb**
## Notes
- Converts ownerid from String to ObjectId for MongoDB
    - Its better to store as ObjectId in MongoDB as it takes up half the space 
- For jackson, add a serializer for ObjectId to be transformed into a String during json serialization (We still want to
return a string representation of ObjectId when serializing object into json)

## Testing
- I tested by adding a rating system "1-10 continuous" on the backend
- I tested by sending a json response for the rating system and confirmed it was a string

### CODE REVIEW

**Review by christopherbarnett — APPROVED** [review]
cb - looked over it and lgtm!!!

**Review by AHuangHe — APPROVED** [review]
LGTM!

File: server/src/main/java/com/github/animelist/animelist/config/JacksonConfig.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class JacksonConfig {
+
+    @Bean
+    public Jackson2ObjectMapperBuilderCustomizer jsonCustomizer() {
+        return builder -> builder.serializerByType(ObjectId.class, new ToStringSerializer());
```
Comment (rowantran) [inline server/src/main/java/com/github/animelist/animelist/config/JacksonConfig.java:14]: the Jackson serializer is used to serialize the object into a JSON document right? so if we're converting the objectID to a string again then how does it save space? not sure I fully understand. code lgtm though

Comment (jvoucsb) [reply server/src/main/java/com/github/animelist/animelist/config/JacksonConfig.java:14]: yes jackson is used to serialize the object into json. we're adding some more configuration to the jackson ObjectMapper SpringBoot already has for us through this configuration and bean file. it's used when we want to map the object to json whenever we want to return a api response.

mongodb does not use springboot's jackson ObjectMapper to convert the object to json. I think it uses another implementation, MongoTypeMapper, to do that which we can change and customize in the "mapping" layer [detailed here](https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#mongo-template.type-mapping)

in short, when mapping the object to store in MongoDB the objectId is correctly stored as an ObjectId. When mapping the object to json for an api response then the ObjectID is turned into a string.

**Review by zihaoxu20010109 — APPROVED** [review]
LGTM!
